unit/sharing_map.cpp

```diff
@@ -20,317 +20,302 @@ void fill2(smt &sm)
   sm.insert("n", "5");
 }
 
-// tests
-
-void sharing_map_interface_test()
+class some_keyt
 {
-  std::cout << "Running interface test" << std::endl;
-
-  // empty map
+public:
+  some_keyt(size_t s) : s(s)
   {
-    smt sm;
-
-    assert(sm.empty());
-    assert(sm.size()==0);
-    assert(!sm.has_key("i"));
   }
 
-  // insert elements
-  {
-    smt sm;
+  size_t s;
 
-    smt::const_find_type r1=sm.insert(std::make_pair("i", "0"));
-    assert(r1.second);
+  bool operator==(const some_keyt &other) const
+  {
+    return s==other.s;
+  }
+};
 
-    smt::const_find_type r2=sm.insert("j", "1");
-    assert(r2.second);
+class some_key_hash
+{
+public:
+  size_t operator()(const some_keyt &k) const
+  {
+    return k.s & 0x3;
+  }
+};
 
-    smt::const_find_type r3=sm.insert(std::make_pair("i", "0"));
-    assert(!r3.second);
 
-    assert(sm.size()==2);
-    assert(!sm.empty());
-  }
 
-  // place elements
+TEST_CASE("sharing map", "[core][util][sharing_map]")
```

Comment:
Unfortunately this file is impossible to diff because of the change in indenting. Can you split the changes into one that does the change and then one that indents it out correctly (or one that indents it out wrongly and then one that does the change)?